test/datahike/test/transact_test.cljc

```diff
@@ -346,15 +346,13 @@
           (is (= (into {} (d/entity @conn tx-id))
                  {:prop4 "prop4"})))))))
 
-; TODO Fix with https://github.com/replikativ/datahike/issues/433
-; and improve the test
-#_(deftest test-tx-meta
-    (let [conn (du/setup-db)
-          tx   (d/transact conn {:tx-data [{:name "Sergey"
-                                            :age  5}]
-                                 :tx-meta {:foo "bar"}})]
-      (is (= (:tx-meta tx)
-             {:foo "bar"}))))
+(deftest test-tx-meta
+  (let [conn (du/setup-db)
+        tx   (d/transact conn {:tx-data [{:name "Sergey"
+                                          :age  5}]
+                               :tx-meta {:foo "bar"}})]
+    (is (= (:tx-meta tx)
```

Comment:
There should be more than this simple test since meta data like `:db/txInstant` is essential for features like the history. What happens when transacting available `:db/txInstant` meta data? What happens if the meta data is not defined in the schema? Can the meta data be found within  `datoms`? Can they be found in the index through queries? Are there differences between schema flexibility? Can meta data be removed again? Can it be changed within a new transaction? Does tx-meta return `:db/txInstant` if it is not transacted by the user?

Reply:
Also please link the issue in the description.

Reply:
Added some more tests now. I hope I did understand your advices, not sure though.